Physical_Engine/Query/Geometry.cs

```diff
@@ -49,8 +49,12 @@ public static ICurve Geometry(this IFramingElement framingElement)
         [Output("surface", "The defining location surface geometry of the ISurface with its openings represented")]
         public static oM.Geometry.ISurface Geometry(this oM.Physical.Elements.ISurface surface)
         {
-            ICurve exBound = (surface.Location as PlanarSurface).ExternalBoundary;
-            List<ICurve> inBound = surface.Openings.Select(o => (o.Location as PlanarSurface).ExternalBoundary).ToList();
+            PlanarSurface planarSurface = surface?.Location as PlanarSurface;
+            if (planarSurface == null)
+                return null;
+
+            ICurve exBound = planarSurface.ExternalBoundary;
+            List<ICurve> inBound = surface?.Openings?.Select(o => (o?.Location as PlanarSurface)?.ExternalBoundary).ToList();
```

Comment:
I gave it a shot to flood it with even more null checks 🙈 
```suggestion
            ICurve exBound = (surface?.Location as PlanarSurface)?.ExternalBoundary;
            if (exBound == null)
                return null;

            List<ICurve> inBound = surface?.Openings?.Select(o => (o?.Location as PlanarSurface)?.ExternalBoundary).Where(x => x != null).ToList();
```